client/src/main/java/io/lionweb/client/delta/InMemoryDeltaChannel.java

```diff
@@ -0,0 +1,53 @@
+package io.lionweb.client.delta;
+
+import io.lionweb.client.delta.messages.*;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.function.Function;
+import org.jetbrains.annotations.Nullable;
+
+public class InMemoryDeltaChannel implements DeltaChannel {
```

Comment:
When implementing C#, I first thought it would be very nice for debugging to have both the client and the server in the same process. It turned out to be _very_ confusing, as we have several instances of the same node (with the same id) to handle.
To me, it now feels more convenient to have them in separate processes.

YMMV, just sharing some pain (-: